ui/src/app/views/account-creation/component.tsx

```diff
@@ -0,0 +1,294 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+import * as fp from 'lodash/fp';
+
+import {fullUrl, handleErrors} from 'app/utils/fetch';
+
+import {
+  CreateAccountRequest,
+  DataAccessLevel,
+  FetchArgs,
+  Profile,
+  ProfileApiFetchParamCreator,
+} from 'generated/fetch/api';
+
+import {
+  Error,
+  ErrorMessage,
+  LongInput,
+  styles as inputStyles
+} from 'app/components/inputs';
+
+import {
+  InfoIcon,
+  ValidationIcon
+} from 'app/components/icons';
+
+import {
+  TooltipTrigger
+} from 'app/components/popups';
+
+import {Button} from 'app/components/buttons';
+import { FormSection } from 'app/components/forms';
+import {BoldHeader} from 'app/components/headers';
+
+function isBlank(s: string) {
+  return (!s || /^\s*$/.test(s));
+}
+
+export interface AccountCreationProps {
+  invitationKey: string;
+  setProfile: Function;
+}
+
+export interface AccountCreationState {
+  profile: Profile;
+  usernameCheckInProgress: boolean;
+  usernameConflictError: boolean;
+  creatingAccount: boolean;
+  showAllFieldsRequiredError: boolean;
+}
+
+export class AccountCreation extends
+  React.Component<AccountCreationProps, AccountCreationState> {
+  private usernameCheckTimeout: NodeJS.Timer;
+
+  constructor(props: AccountCreationProps) {
+    super(props);
+    this.state = {
+      profile: {
+        // Note: We abuse the "username" field here by omitting "@domain.org". After
+        // profile creation, this field is populated with the full email address.
+        username: '',
+        dataAccessLevel: DataAccessLevel.Unregistered,
+        givenName: '',
+        familyName: '',
+        contactEmail: '',
+        currentPosition: '',
+        organization: '',
+        areaOfResearch: '',
+      },
+      usernameCheckInProgress: false,
+      usernameConflictError: false,
+      creatingAccount: false,
+      showAllFieldsRequiredError: false
+    };
+  }
+
+  createAccount(): void {
+    const {invitationKey, setProfile} = this.props;
+    this.setState({showAllFieldsRequiredError: false});
+    const requiredFields =
+      [this.state.profile.givenName, this.state.profile.familyName,
+        this.state.profile.username, this.state.profile.contactEmail,
+        this.state.profile.currentPosition, this.state.profile.organization,
+        this.state.profile.areaOfResearch];
+    if (requiredFields.some(isBlank)) {
+      this.setState({showAllFieldsRequiredError: true});
+      return;
+    } else if (this.isUsernameValidationError()) {
+      return;
+    }
+    const request: CreateAccountRequest = {
+      profile: this.state.profile,
+      invitationKey: invitationKey
+    };
+    this.setState({creatingAccount: true});
+    const args: FetchArgs = ProfileApiFetchParamCreator().createAccount(request);
+    fetch(fullUrl(args.url), args.options)
```

Comment:
FYI: My API client PR has been merged: https://github.com/all-of-us/workbench/pull/1676